<?xml version="1.0"?>
<launch>
    <!-- Arguments -->
    <arg name="pybullet_gui" default="false"/>
    <arg name="paused" default="false"/>

    <arg name="use_front_iiwa" default="true"/>
    <arg name="use_back_iiwa" default="true"/>
    <arg name="simulation" default="true" />
    <arg name="model_list" default="['iiwas_frame', 'iiwa_front', 'iiwa_back']"/>

    <!-- Description -->
    <include file="$(find iiwas_description)/launch/iiwas_description.launch">
        <arg name="use_front_iiwa" value="$(arg use_front_iiwa)" />
        <arg name="use_back_iiwa" value="$(arg use_back_iiwa)" />
    </include>

    <node name="rviz" pkg="rviz" type="rviz"
          args="-d $(find iiwas_description)/config/iiwas_two_separate.rviz"/>

    <group ns="iiwa_front">
        <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
            <param name="publish_frequency" type="double" value="50.0"/>
        </node>
    </group>

    <group ns="iiwa_back">
        <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
            <param name="publish_frequency" type="double" value="50.0"/>
        </node>
    </group>

    <rosparam file="$(find iiwas_pybullet)/config/iiwa.config" command="load" />

    <!-- Simulation -->
    <include file="$(find pybullet_ros)/ros/launch/empty_world.launch">
        <arg name="config_file" value="$(find iiwas_pybullet)/config/environment.config" />
        <arg name="environment" default="iiwas_pybullet.plugins.env_plugin" />
        <arg name="pybullet_gui" default="$(arg pybullet_gui)" /> <!-- if true pybullet gui will show -->
        <arg name="pause_simulation" default="$(arg pause_simulation)" /> <!-- if true, will boot with paused physics -->
        <arg name="parallel_plugin_execution" default="False" />
        <arg name="use_deformable_world" default="False" /> <!-- enable/disable soft body simulation -->
        <arg name="gui_options" default="" />
        <arg name="model_list" value="$(arg model_list)"/>
    </include>


    <group ns="iiwas_frame">
        <node pkg="robot_state_publisher" type="robot_state_publisher" name="frame_publisher" output="screen">
            <param name="publish_frequency" type="double" value="50.0"/>
        </node>
    </group>

</launch>

